src/main/java/seedu/address/model/appointment/AppointmentList.java

```diff
@@ -0,0 +1,163 @@
+        requireAllNonNull(target, editedAppointment);
+
+        int index = internalList.indexOf(target);
+        internalList.set(index, editedAppointment);
+    }
+
+    /**
+     * Removes an appointment from the list.
+     *
+     * @param toRemove The appointment to be removed.
+     */
+    public void remove(Appointment toRemove) {
+        requireNonNull(toRemove);
+        internalList.remove(toRemove);
+    }
+
+    /**
+     * Sets the entire list of appointments to a new list of appointments.
+     *
+     * @param replacement The new list of appointments.
+     */
+    public void setAppointments(AppointmentList replacement) {
+        requireNonNull(replacement);
+        internalList.setAll(replacement.internalList);
+    }
+
+    /**
+     * Sets the entire list of appointments to a new list of appointments.
+     *
+     * @param appointments The new list of appointments.
+     */
+    public void setAppointments(List<Appointment> appointments) {
+        requireAllNonNull(appointments);
+        internalList.setAll(appointments);
+    }
+
+    /**
+     * Retrieves an unmodifiable view of the internal list of appointments.
+     *
+     * @return An unmodifiable observable list of appointments.
+     */
+    public ObservableList<Appointment> asUnmodifiableObservableList() {
+        return internalUnmodifiableList;
+    }
+
+    /**
+     * Returns an iterator over the appointments in the list.
+     *
+     * @return An iterator of appointments.
+     */
+    @Override
+    public Iterator<Appointment> iterator() {
+        return internalList.iterator();
+    }
+
+    /**
+     * Checks if this appointment list is equal to another object.
+     *
+     * @param other The object to compare to.
+     * @return {@code true} if the objects are equal, {@code false} otherwise.
+     */
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        if (!(other instanceof AppointmentList)) {
+            return false;
+        }
+
+        AppointmentList otherAppointmentList = (AppointmentList) other;
+        return internalList.equals(otherAppointmentList.internalList);
+    }
+
+    /**
+     * Returns a hash code value for the appointment list.
+     *
+     * @return The hash code.
+     */
+    @Override
+    public int hashCode() {
+        return internalList.hashCode();
+    }
+
+    /**
+     * Returns a string representation of the appointment list.
+     *
+     * @return A string representation of the list of appointments.
+     */
+    @Override
+    public String toString() {
+        return internalList.toString();
+    }
+
+    /**
+     * Checks if the appointments in a list are unique.
+     *
+     * @param appointments The list of appointments to check.
+     * @return {@code true} if all appointments are unique, {@code false} otherwise.
+     */
+    private boolean appointmentsAreUnique(List<Appointment> appointments) {
```

Comment:
as mentioned earlier, i think it is more valuable to be checking for time overlaps between appointments here